enhancements/authentication/direct-use-of-oidc.md

```diff
@@ -0,0 +1,450 @@
+
+    // UsernameClaim - The OpenID claim to use as the user name. Note that claims
+    // other than the default ('sub') is not guaranteed to be unique and immutable.
+    // This flag is experimental, please see the authentication documentation for
+    // further details.
+    // TODO: what is the authentication documentation?
+    UsernameClaim string `json:"usernameClaim,omitempty"`
+
+    // UsernamePrefix - If provided, all usernames will be prefixed with this
+    // value. If not provided, username claims other than 'email' are prefixed
+    // by the issuer URL to avoid clashes. To skip any prefixing, provide the
+    // value '-'.
+    UsernamePrefix string `json:"usernamePrefix,omitempty"`
+
+    // GroupsClaim - If provided, the name of a custom OpenID Connect claim for
+    // specifying user groups. The claim value is expected to be a string or
+    // array of strings. This flag is experimental, please see the authentication
+    // documentation for further details.
+    GroupsClaim string `json:"groupsClaim,omitempty"`
+
+    // GroupsPrefix - If provided, all groups will be prefixed with this value to
+    // prevent conflicts with other authentication strategies.
+    GroupsPrefix string `json:"groupsPrefix,omitempty"`
+
+    // SigningAlgorithms - List of allowed JOSE asymmetric signing
+    // algorithms. JWTs with a supported 'alg' header values are:
+    // RS256, RS384, RS512, ES256, ES384, ES512, PS256, PS384, PS512.
+    // Values are defined by RFC 7518 https://tools.ietf.org/html/rfc7518#section-3.1.
+    //
+    // Default value: RS256
+    // TODO: define an enum type for this?
+    SigningAlgorithms []string `json:"signingAlgorithms,omitempty"`
+    
+    // CAFile - If set, the OpenID server's certificate will be verified by one
+    // of the authorities in the CAFile, otherwise the host's root CA
+    // set will be used.
+    CAFile ConfigMapNameReference `json:"caFile,omitempty"`
+}
+
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+There will be a new observer in `cluster-kube-apiserver-operator` which is going
+to observe the `KubeOIDC` field of the `authentication.config.openshift.io/cluster`
+configuration custom resource:
+- If the `KubeOIDCConfiguration` feature gate is
+enabled, the observer is going to turn the `KubeOIDC` configuration into
+`kube-apiserver` flags.
+- If the `KubeOIDCConfiguration` is disabled, any previous `kube-apiserver` OIDC-related
+  flags get removed from the configuration and the `KubeOIDC` field of the 
+  `authentication.config.openshift.io/cluster` resource is ignored
+
+Additionally to the above, the configmap from `authentication.config.openshift.io/cluster`
+`KubeOIDC.CAFile` should get synchronized from the `openshift-config` namespace
+to `openshift-kube-apiserver` namespace if the `KubeOIDCConfiguration` feature
+gate is enabled.
+
+// TODO: decide how involved should the `oauth-server` be when OIDC is configured
```

Comment:
this cannot be a TODO, it must be resolved before we mark this as implementable. We *will* face questions.  Given previously stated goals, I suggest co-existence as an anti-goal and disabling the existing as a goal.